tensorflow/lite/micro/kernels/split_v.cc

```diff
@@ -0,0 +1,173 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/lite/c/builtin_op_data.h"
+#include "tensorflow/lite/c/common.h"
+#include "tensorflow/lite/kernels/internal/tensor_ctypes.h"
+//#include "tensorflow/lite/kernels/internal/tensor.h"
+#include "tensorflow/lite/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/kernels/kernel_util.h"
+#include "tensorflow/lite/kernels/op_macros.h"
+
+namespace tflite {
+namespace ops {
+namespace micro {
+namespace split_v {
+
+struct OpContext {
+  OpContext(TfLiteContext* context, TfLiteNode* node) {
+    params = reinterpret_cast<TfLiteSplitVParams*>(node->builtin_data);
+    input = tflite::micro::GetEvalInput(context, node, 0);
+    size_splits = tflite::micro::GetEvalInput(context, node, 1);
+    axis = tflite::micro::GetEvalInput(context, node, 2);
+  }
+  TfLiteSplitVParams* params;
+  const TfLiteEvalTensor* input;
+  const TfLiteEvalTensor* size_splits;
+  const TfLiteEvalTensor* axis;
+};
+
+template <typename T>
+TfLiteStatus SplitImpl(TfLiteContext* context, TfLiteNode* node,
+                       OpContext* op_context) {
+  const int outputs_count = NumOutputs(node);
+  const TfLiteEvalTensor* input = tflite::micro::GetEvalInput(context, node, 0);
+  
+  const TfLiteIntArray* input_dims = input->dims;
+  const TfLiteEvalTensor* output0 =     tflite::micro::GetEvalOutput(context, node, 0);
+  const TfLiteIntArray* output_dims = output0->dims;
+  int axis_value = op_context->axis->data.i32[0];
+  
+  const int split_dimensions = input_dims->size;
+  int axis = axis_value < 0 ? axis_value + split_dimensions : axis_value;
```

Comment:
can you add a comment about why this is needed? A lot of such code on mobile is to support dynamic resizing etc. which are unsupported on micro. If this is not needed for your application, I would recommend adding a TFLITE_DCHECK(axis_value >= 0); so that a model with a negative axis_value will fail at a predictable location.